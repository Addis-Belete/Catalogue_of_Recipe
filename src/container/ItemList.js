/* eslint-disable no-console */
/* eslint-disable react/forbid-prop-types */
import React, { useEffect, useState } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import Recipe from '../components/Recipe';
import { fetchRecipes } from '../action';
import styles from '../style/style.module.css';
import { fetchIngredients } from '../action/ingredient';
import CategoryFilter from '../components/Category';

const ItemList = ({ recipes, fetchRecipes }) => {
  console.log(recipes);
  const [fil, setFil] = useState('');
  useEffect(() => {
    fetchRecipes();
    fetchIngredients();
  }, []);

  const search = (recipes) => recipes.recipes.meals.filter((result) => result.strMeal
    .toLowerCase().indexOf(fil) > -1);

  return (

    <div>
      <div className="category">

        <input
          className={styles.input}
          type="text"
          placeholder="Search for recipes"
          value={fil}
          onChange={(e) => setFil((e.target.value).toLowerCase())}
        />
        <CategoryFilter />
      </div>

      <Recipe recipes={search(recipes)} />
    </div>
  );
};
export const mapStateToProps = (state) => ({
  recipes: state.recipe,

});

export const mapDispatchToProps = (dispatch) => ({
  fetchRecipes: () => dispatch(fetchRecipes()),
  fetchIngredient: () => dispatch(fetchIngredients()),
});

ItemList.propTypes = {
  recipes: PropTypes.object,
  fetchRecipes: PropTypes.func.isRequired,

};
ItemList.defaultProps = {
  recipes: {},

};

export default connect(mapStateToProps, mapDispatchToProps)(ItemList);
