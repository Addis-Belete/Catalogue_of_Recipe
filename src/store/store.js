import { applyMiddleware, createStore } from 'redux';
import thunk from 'redux-thunk';
import { composeWithDevTools } from 'redux-devtools-extension';
import { recipeReducer } from '../reducer';

const store = createStore(recipeReducer, composeWithDevTools(applyMiddleware(thunk)));
export default store;
