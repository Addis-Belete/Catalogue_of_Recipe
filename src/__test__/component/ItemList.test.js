import React from 'react';
import { render, cleanup, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import ItemList from '../../container/ItemList';
import store from '../../store/store';

afterEach(cleanup);

it('should render the list of ingredients', () => {
  const { asFragment } = render(
    <Provider store={store}>
      <BrowserRouter>
        <ItemList />
      </BrowserRouter>
    </Provider>,
  );
  expect(
    asFragment(
      <Provider store={store}>
        <ItemList />
      </Provider>,
    ),
  ).toMatchSnapshot();
});
