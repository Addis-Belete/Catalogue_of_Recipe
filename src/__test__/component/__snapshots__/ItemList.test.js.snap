// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the list of ingredients 1`] = `
<DocumentFragment>
  <div>
    <input
      class="input"
      placeholder="Search for recipes"
      type="text"
      value=""
    />
    <div>
      <div
        class="tablet desktop"
      >
        <div>
          <a
            href="/2020"
          >
            <img
              alt="recipeImage"
              class="img"
              src="undefined"
            />
            <p
              class="p"
            >
              hello world
            </p>
          </a>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
